### ISSUE TITLE
Sim Screens and Sim Resources should be seperate regions

### ISSUE THREAD

**Issue by terracoda** [phetsims/joist#641]
While reviewing implementation of RaP sim, I noticed that the SIm Screens region and the SIm Resources Regions are combined.

I think this generalizes the implementation, and for the non-visual experience it makes no difference if the Sim Screen buttons are in a region before the Sim Resources. 

Here's the code  from the HomeScree.md file

```html
<!-- Navigation landmark section for Sim Screens, that includes all relevant screen buttons -->
<nav aria-lablledby="labelSimScreens">
  <h2 id="labelSimScreens">Sim Screens</h2>
	 <!-- All relevant screen buttons without aria-describedby. -->
	 <ol>
	   <li><button>{{Atom}} Screen</button>
		   <p id="descriptionScreen01">{{Explore what makes up an atom}}.</p></li>
	   <li><button>{{Symbol}} Screen</button>
		   <p id="descriptionScreen02">{{Investigate atoms and their atomic symbols}}.</p></li>
	   <li><button>{{Game}} Screen</button>
		   <p id="descriptionScreen03">{{Test your knowledge of atoms and their atomic symbols}}.</p></li>
	 </ol>
</nav>

<!-- Section for Sim Resources, that includes all relevant buttons -->
<section aria-labeledby="lableSimRsources">
  <h2 id="lableSimRsources">Sim Resources</h2>
  <!-- Mute Sound, Keyboard Shortcuts, PhET Menu  -->
<section>
```
I'd like to try a `nav` element for the list of Sim Screens because it seems like navigation to me, but we could potentially use a `section` element if the `nav` element doesn't work out.

Both the `nav` element and `section` elements are landmark regions, thus easy to find with screen reader commands.

**Comment by terracoda** [comment]
This issue is related to https://github.com/phetsims/joist/issues/496

**Comment by terracoda** [comment]
We can use the same heading structure for the Home Screen and each Sim Screen if we keep the regions separate
![Screen Shot 2020-07-16 at 11 36 25 AM](https://user-images.githubusercontent.com/1902116/87682073-fe425c00-c759-11ea-8174-3f131e719504.png)
![Screen Shot 2020-07-16 at 11 07 25 AM](https://user-images.githubusercontent.com/1902116/87682578-8f193780-c75a-11ea-9b2c-71fc52b292d7.png)

**Comment by terracoda** [comment]
@zepumph the names for the buttons on the Home screen should automatically populate the names for the buttons on the each individual Screen.

This might be more straightforward once the sim screens are in their own region.

**Comment by terracoda** [comment]
One other thing that is not implemented on the Home Screen is `aria-describedby` for the Screen buttons.

I think it would be helpful read out the little screen descriptions automatically on the Home Screen, but not on the individual Sim Screens. On the individual screens,  the description can be explored on-demand.

You would need logic that said, if Home Screen, use `aria-describedby` on sim screen buttons.

**Comment by zepumph** [comment]
I poked at this just now. I will keep going next week.

```diff
Index: joist-strings_en.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- joist-strings_en.json	(revision e12b7b2f00fe9a1e540032e9d5ecb1b1f8664721)
+++ joist-strings_en.json	(date 1595383906971)
@@ -188,7 +188,7 @@
     "homeScreenDescriptionPattern": {
       "value": "Come explore with {{name}}. It has {{screens}} screens."
     },
-    "simResourcesAndTools": {
+    "simResources": {
       "value": "Sim Resources"
     },
     "simScreensResourcesAndTools": {
Index: js/NavigationBar.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/NavigationBar.js	(revision e12b7b2f00fe9a1e540032e9d5ecb1b1f8664721)
+++ js/NavigationBar.js	(date 1595384150949)
@@ -45,7 +45,7 @@
 const simTitleWithScreenNamePatternString = joistStrings.simTitleWithScreenNamePattern;
 
 const simScreensResourcesAndToolsString = joistStrings.a11y.simScreensResourcesAndTools;
-const simResourcesAndToolsString = joistStrings.a11y.simResourcesAndTools;
+const simResourcesString = joistStrings.a11y.simResources;
 const simScreensString = joistStrings.a11y.simScreens;
 
 // constants
@@ -88,7 +88,7 @@
     labelTagName: 'h2',
 
     // Use a different string, omitting "screens" if single screen sim.
-    labelContent: this.simScreens.length === 1 ? simResourcesAndToolsString : simScreensResourcesAndToolsString
+    labelContent: this.simScreens.length === 1 ? simResourcesString : simScreensResourcesAndToolsString
   } );
 
   // @private - The nav bar fill and determining fill for elements on the nav bar (if it's black, the elements are white)
@@ -318,6 +318,18 @@
       buttons.setVisible( !showHomeScreen );
     }
   } );
+
+  // a11y layout
+  const simResourcesWrapper = new Node( {
+    containerTagName: 'section',
+    labelTagName: 'h2',
+    labelContent: simResourcesString,
+    ariaLabelledbyAssociations: [ {
+      thisElementName: PDOMPeer.CONTAINER_PARENT,
+      otherNode: this,
+      otherElementName: PDOMPeer.LABEL_SIBLING,
+    } ]
+  } );
 }
 
 joist.register( 'NavigationBar', NavigationBar );

```

**Comment by zepumph** [comment]
@terracoda, is everything discussed above only for the home screen? So would it still say "Sim Screens, Resources, and Tools" for the nav bar when on the "Explore Screen"?

**Comment by zepumph** [comment]
RE: the first photo in https://github.com/phetsims/joist/issues/641#issuecomment-659445273, support for the help text has already been implemented, it is just that it defaults to "{{Screen}} Screen" when not provided (which I hadn't added it yet to RaP. No changes needed here in joist.

- [ ] Would you prefer a different default?

**Comment by zepumph** [comment]
Above I created a new Property in Screen.js which is the display name to be used in PDOM description. I am linking to that everywhere we need to set the screen name so that if it ever changes (as it can with PhET-iO), the description will stay in sync. I prefer this slightly to listening to the nameProperty and doing a fillIn at all usage sites.

Up next I will fix the home screen labels, and then I will work on regions/headings.

**Comment by zepumph** [comment]
Ok, now when the nameProperty changes, the following change:
* Visual display on home screen
* Label on home screen (PDOM)
* visual screen button on nav bar
* label for screen button on nav bar (PDOM)
* main sim H1 in PDOM
* intro screen summary sentence in PDOM

Next is regions/headings

**Comment by zepumph** [comment]
In the above commits. . . 
* I separated out the "Sim Screens" from the "Sim Resources"
* Removed the duplication on the Home screen of having "Sim Screens" AND "Sim Screens, Resources, and Tools"
* I update the aria-labelledby associtaions of both the nav and section regions. 
* I simplified the previous code quite a bit by inlining strings and getting rid of extra intermediate Nodes.


I think this issue is all complete. @terracoda please review.

**Comment by samreid** [comment]
The sim fails on startup when launching with `?homeScreen=false`.

**Comment by terracoda** [comment]
@zepumph, on Voice Over the Home Screen Sounds really good to me!
I think you got everything that I requested.

**Comment by terracoda** [comment]
I am un-assigning myself. Reassign if you need me again.

**Comment by zepumph** [comment]
This was fixed above, and I added some CT tests over in https://github.com/phetsims/aqua/issues/105 to make sure this doesn't happen again. Thanks @samreid. 

Thanks @terracoda for the review.

@samreid, would you please review the above commits in joist? I am especially interested in your thoughts about me making a separate Property for the PDOM display name of the screen. This felt clean in some ways, but annoying to have two Properties. What do you think about that and the whole changeset?

**Comment by samreid** [comment]
Would it be better to only create the `HomeButton` if `sim.homeScreen` is defined?  It would have to be done in a phet-io-friendly way.

**Comment by zepumph** [comment]
I think there are more concerns that just PhET-iO, as much of the nav bar layout takes into account the homeButton even if it isn't displayed. It seems like that helps have a consistent look whether `?homeScreen=false` is provided or not (though I'm unsure how important that is). 

I think I would prefer to create it each time as it is being done now.

**Comment by samreid** [comment]
I tried creating `new Rectangle(0,0,0,0)` for layout if there is no home screen button in this patch:

<details>

```diff
Index: js/NavigationBar.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/NavigationBar.js	(revision adfe94398930d77932a3c27e457ce0e12df9174e)
+++ js/NavigationBar.js	(date 1597098815766)
@@ -25,7 +25,6 @@
 
 import DerivedProperty from '../../axon/js/DerivedProperty.js';
 import Property from '../../axon/js/Property.js';
-import StringProperty from '../../axon/js/StringProperty.js';
 import Dimension2 from '../../dot/js/Dimension2.js';
 import inherit from '../../phet-core/js/inherit.js';
 import StringUtils from '../../phetcommon/js/util/StringUtils.js';
@@ -199,11 +198,12 @@
     buttons.setVisible( false );
     this.barContents.addChild( buttons );
 
-    // @private - Create the home button
-    this.homeButton = new HomeButton(
+    // @private {HomeButton|Rectangle} - Create the home button, or an empty rectangle (for layout) if the home button
+    // should not be displayed
+    this.homeButton = sim.homeScreen ? new HomeButton(
       NAVIGATION_BAR_SIZE.height,
       sim.lookAndFeel.navigationBarFillProperty,
-      sim.homeScreen ? sim.homeScreen.pdomDisplayNameProperty : new StringProperty( 'NO HOME SCREEN' ),
+      sim.homeScreen.pdomDisplayNameProperty,
       tandem.createTandem( 'homeButton' ), {
         listener: () => {
           sim.screenProperty.value = sim.homeScreen;
@@ -213,7 +213,7 @@
             sim.homeScreen.view.focusHighlightedScreenButton();
           }
         }
-      } );
+      } ) : new Rectangle( 0, 0, 0, 0 );
 
     // Add the home button, but only if the homeScreen exists
     sim.homeScreen && buttons.addChild( this.homeButton );

```

</details>

This seemed to work well.  In the case of long strings, it allows more room for screen icons (instead of reserving a blank space):

Master:
![image](https://user-images.githubusercontent.com/679486/89838231-f216a800-db27-11ea-91bb-84b37f9c0cc4.png)

The Patch:
![image](https://user-images.githubusercontent.com/679486/89838248-fc38a680-db27-11ea-98d6-96a0725cd180.png)

It is a question for designers which is preferable.  But even if designers recommend reserving that space, I recommend it be done with `new Rectangle(42,38)` instead of an entire HomeButton.  Here's the same patch but with (42,38)

![image](https://user-images.githubusercontent.com/679486/89838439-7832ee80-db28-11ea-9d70-21609c32b6c7.png)

Looking forward to discussing further.  Should we reach out to designers?

**Comment by zepumph** [comment]
I don't think that the extra spacing will pass designer code review. It seems like there are two options here:

1. (the way it is now) Create the HomeButton eagerly, even when not shown. This way it can work well with a consistent layout and with PhET-iO.
2. When there is no home screen, swap it out for a Rectangle of the same dimensions. This will most likely involve converting a HomeButton to use a PhetioCapsule.

I prefer option 1, given its uses when it is not displayed. From this issue, I have had to re-remember how PhET-iO instrumentation should be done on a "platform" level, even when some runtimes may never have access to a component, it should still be instrumented because the sim as a "platform" still supports an instrumented homeScreen. Is that correct?

**Comment by samreid** [comment]
The most defensive strategy would be to avoid creating the HomeButton when it will not be used.  The code seems clearer and more maintainable that way.  I'm imagining future maintainers running into this code and wondering "why are we creating a HomeButton if it will never be used"?  It also uses CPU and heap to create a HomeButton.  I think treating the simulation like a platform only applies when there are no query parameters.  It is natural that ?screens=2 or ?homeScreen=false would eliminate phetioIDs from the runtime.

My main problem with the `new Rectangle(0,0,42,38)` patch from above is how to keep those dimensions in sync with the home button dimensions.  Perhaps an assertion when we do create the `HomeButton` that it matches a constant for the spacer dimension?  Then if we resize the home button without resizing the spacer we would get an assertion error?

**Comment by zepumph** [comment]
> I think treating the simulation like a platform only applies when there are no query parameters. It is natural that ?screens=2 or ?homeScreen=false would eliminate phetioIDs from the runtime.

That is not how we handled the creation of blocks in EFAC. There beakers and blocks had to be turned into PhetioGroups because query parameters could alter the startup configuration. This was dynamic to the platform, but not to the runtime. I think we would need to do that here also with a `PhetioCapsule` if we choose to conditionally create the HomeButton. I am ok (not more) with this, but feel like often you have been a proponent for creating everything eagerly when possible to simplify PhET-iO. What are you feeling here?

**Comment by samreid** [comment]
Thanks, the blocks example in EFAC is very helpful.  In that case, we wouldn't want to disable all validation just because a different block starting configuration was created.  I guess I'm confused about that though, because when we run with ?screens=... we do disable validation.  So is `?screens=...` a special case?  This issue can probably be closed (or put on hold) but I'd like to discuss this at an upcoming Wednesday meeting so I can understand better.

**Comment by samreid** [comment]
@chrisklus @zepumph and I discussed this today:

Either we need to create one “API” per combination of query parameters, or make sure that any combination of query parameters uses the exact same API.  Since it would be too difficult to create one "API" for each combination of query parameters, we choose to just have one API and the query parameters work within it.

We have an edge case for screens, because we know that is a well-defined subset.

To make sure that APIs with query parameters are trackable/testable, we should treat the whole sim as an API, and only try to subset for ?screens.
